docker/agent/Dockerfile

```diff
@@ -44,18 +44,19 @@ RUN apt-get update \
 RUN curl -LO https://packages.microsoft.com/keys/microsoft.asc && \
     echo 2cfd20a306b2fa5e25522d78f2ef50a1f429d35fd30bd983e2ebffc2b80944fa microsoft.asc| sha256sum --check - && \
     apt-key add microsoft.asc && \
-    apt-add-repository https://packages.microsoft.com/debian/11/prod && \
+    rm microsoft.asc && \
+    echo deb https://packages.microsoft.com/debian/12/prod bookworm main >> /etc/apt/sources.list.d/microsoft.list && \
     apt-get update && \
     apt-get install -y libmsquic && \
-    rm -rf /var/lib/apt/lists/* microsoft.asc
```

Comment:
Why?

Reply:
Nvm

Reply:
It moved above, but I'd also like to understand why.

Reply:
I took these from https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/main/src/debian/12/helix/amd64/Dockerfile#L41-L49

I don't know if that's necessary